src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1345,6 +1356,84 @@ void QgsOgrProvider::loadFields()
       mDefaultValues.insert( createdFields, defaultValue );
     }
 
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,3,0)
+    if ( const char *domainName = OGR_Fld_GetDomainName( fldDef ) )
+    {
+      // dataset retains ownership of domain!
+      if ( OGRFieldDomainH domain = GDALDatasetGetFieldDomain( ds, domainName ) )
+      {
+        switch ( OGR_FldDomain_GetDomainType( domain ) )
+        {
+          case OFDT_CODED:
+          {
+            QVariantList valueConfig;
+            const OGRCodedValue *codedValue = OGR_CodedFldDomain_GetEnumeration( domain );
+            while ( codedValue && codedValue->pszCode )
+            {
+              const QString code( codedValue->pszCode );
+              const QString value( codedValue->pszValue );
```

Comment:
In GeoPackage, in odd cases, you could potentially have the pszValue to be NULL (not sure if that might happen in practice, but the spec requirements don't forbid this. This could be used to restrict a field to a set of values that aren't necessary coded. So perhaps in that case replicate the pszCode value ?

Reply:
Fixed